ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/LocalTxSubmission/Server.hs

```diff
@@ -30,13 +31,10 @@ localTxSubmissionServer tracer mempool =
     server = LocalTxSubmissionServer {
       recvMsgSubmitTx = \tx -> do
         traceWith tracer $ TraceReceivedTx tx
-        res <- addLocalTxs mempool [tx]
-        case res of
-          [addTxRes] -> case addTxRes of
-            MempoolTxAdded _tx             -> return (SubmitSuccess, server)
-            MempoolTxRejected _tx addTxErr -> return (SubmitFail addTxErr, server)
-          -- The output list of addTxs has the same length as the input list.
-          _                 -> error "addTxs: unexpected result"
+        I addTxRes <- addLocalTxs mempool (I tx)
```

Comment:
If we wouldn't have to support GHC 8.10, I think [`Solo`](https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-Tuple.html#t:Solo) would be more idiomatic here.

Reply:
I would note this down on a comment here. For future action.